docs/dialect/graphblas_dialect_tutorials/graphblas_optimize/fuse_select.ipynb

```diff
@@ -0,0 +1,340 @@
+      "    %0:3 = graphblas.matrix_select %arg0 {selectors = [\"tril\", \"triu\", \"gt0\"]} : tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ \"dense\", \"compressed\" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>> to tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ \"dense\", \"compressed\" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>>, tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ \"dense\", \"compressed\" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>>, tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ \"dense\", \"compressed\" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>>\n",
+      "    return %0#2, %0#1, %0#0 : tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ \"dense\", \"compressed\" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>>, tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ \"dense\", \"compressed\" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>>, tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ \"dense\", \"compressed\" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>>\n",
+      "  }\n",
+      "}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(output_mlir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "450ca5d1",
+   "metadata": {},
+   "source": [
+    "This is difficult to read due to the verbosity of the sparse tensor encodings. Let's make it more terse. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "a3ba8f1c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "module  {\n",
+      "  func @select_fuse_single(%arg0: tensor<?x?xf64, #CSR64>) -> (tensor<?x?xf64, #CSR64>, tensor<?x?xf64, #CSR64>, tensor<?x?xf64, #CSR64>) {\n",
+      "    %0:3 = graphblas.matrix_select %arg0 {selectors = [\"tril\", \"triu\", \"gt0\"]} : tensor<?x?xf64, #CSR64> to tensor<?x?xf64, #CSR64>, tensor<?x?xf64, #CSR64>, tensor<?x?xf64, #CSR64>\n",
+      "    return %0#2, %0#1, %0#0 : tensor<?x?xf64, #CSR64>, tensor<?x?xf64, #CSR64>, tensor<?x?xf64, #CSR64>\n",
+      "  }\n",
+      "}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "output_mlir = tersify_mlir(output_mlir)\n",
+    "print(output_mlir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2c6c42da",
+   "metadata": {},
+   "source": [
+    "As shown above, `--graphblas-optimize` combined the original 3 uses of `graphblas.matrix_select` into one!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3510cff0",
+   "metadata": {},
+   "source": [
+    "## Fusing Single Selector or Multi-Selector graphblas.matrix_select Ops\n",
```

Comment:
This 2nd section may not be needed. Most users would expect this to work same as above.

The 3rd section is useful in that it shows how fusion is intelligent in inspecting not just sequences of calls, but the inputs as well to make sure the end result yields a correct answer.

Reply:
I like being more explicit in showing that the fusion also works in the case where we have both single selector and multi-selector ops, so I chose to get rid of the first section and make the second section the primary example. I don't think it's much more complex and gets the idea across. 

I implemented this in https://github.com/metagraph-dev/mlir-graphblas/pull/85/commits/b8074dcc2de4c2509dec1c78e27e048839cba80c.

Let me know what you think.